rocketpy/Components.py

```diff
@@ -0,0 +1,59 @@
+__author__ = "Mateus Stano Junqueira"
+__copyright__ = "Copyright 20XX, RocketPy Team"
+__license__ = "MIT"
+
+from collections import namedtuple
+
+
+class Components:
+    """A group of components, each of which has a position"""
+
+    def __init__(self):
+        self.component_position = namedtuple("component_tuple", "component position")
+        self._components = []
+
+    def __repr__(self):
+        return repr(self._components)
+
+    def __len__(self):
+        return len(self._components)
+
+    def __getitem__(self, index):
+        return self._components[index]
+
+    def __iter__(self):
+        return iter(self._components)
+
+    def add(self, component, position):
+        """Add a component to the list of components"""
+        self._components.append(self.component_position(component, position))
+
+    def get_by_type(self, component_type):
+        """Get the component of a specified type"""
+        component_type_list = [
+            c.component
+            for c in self._components
+            if isinstance(c.component, component_type)
+        ]
+        return (
+            component_type_list[0]
+            if len(component_type_list) == 1
+            else component_type_list
+        )
```

Comment:
I remember having already discussed it, this is better than the earlier approach but still haunts me a little the idea that the return type of this method if undefined. Could you summarize the main arguments favoring this implementation?

An idea could be to separate the method into `get_main_by_type` and `get_all_by_type` (with a docstring that the main component is defined as the first added) or `get_component` and `get_components`.

Reply:
Addresed in https://github.com/RocketPy-Team/RocketPy/pull/355/commits/49556de1b8118d089ce7071430c84370915de9ca